src/storage/object.ts

```diff
@@ -575,32 +574,36 @@ export class ObjectStorage {
           prevPrefix = currPrefix
           delimitedResults.push({
             id: null,
-            name: options?.encodingType === 'url' ? encodeURIComponent(currPrefix) : currPrefix,
+            name: currPrefix,
             bucket_id: object.bucket_id,
           })
           continue
         }
 
-        delimitedResults.push({
-          ...object,
-          name: options?.encodingType === 'url' ? encodeURIComponent(object.name) : object.name,
-        })
+        delimitedResults.push(object)
       }
-      results = delimitedResults
+      searchResult = delimitedResults
     }
 
     let isTruncated = false
 
-    if (results.length > limit) {
-      results = results.slice(0, limit)
+    if (searchResult.length > limit) {
+      searchResult = searchResult.slice(0, limit)
       isTruncated = true
     }
 
-    const folders = results.filter((obj) => obj.id === null)
-    const objects = results.filter((obj) => obj.id !== null)
+    const folders: Obj[] = []
+    const objects: Obj[] = []
+    searchResult.forEach((obj) => {
```

Comment:
I'm wondering if it's better to not have a double loop to URL encode, and instead simply urlDecode the last item name, when encodingType === 'url'

Reply:
This single loop replaces the two filters (loops) that used to be here. We have to bucket them into folders/objects either way. This just uses that opportunity to do the url encoding.

Reply:
Make sense! Thanks for clarifying